### PR TITLE
Ultra Extended L1b Front Coordinates 

### PR DESCRIPTION
**Pull request by laspsandoval**
# Change Summary

## Overview
The position of the particle along the front of the sensor.

## New Files
- l1b_extended_config.py
   - place to store constants
- ultra_l1b_extended.py
   - The position of the particle along the front of the sensor (xf, yf)
- FM45_40P_Phi28p5_BeamCal_LinearScan_phi28.50_theta-0.00_Ultra_Image_Raw_Event_20240207T102746_withFSWcalcs.csv
    - Test data

## Updated Files
- conftest.py
   - path of test data
   - creation of event data
- lookup_utils.py
   - added ability to lookup array

## Testing
- test_ultra_l1b_extended.py
   - test ultra_l1b_extended.py

### CODE REVIEW

File: imap_processing/tests/ultra/unit/test_ultra_l1b_extended.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    events_fsw_comparison_theta_0,
+):
+    """Tests get_front_y_position function."""
+
+    df = pd.read_csv(events_fsw_comparison_theta_0)
+    df_filt = df[df["StartType"] != -1]
+
+    d, yf = get_front_y_position(
+        de_dataset["START_TYPE"].data, df_filt.Yb.values.astype("float")
+    )
```
Comment (laspsandoval) [inline imap_processing/tests/ultra/unit/test_ultra_l1b_extended.py:40]: Note: there is no test data for the value of d.

File: imap_processing/ultra/l1b/ultra_l1b_extended.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    start_position_tdc : np.array
+        Start Position Time to Digital Converter (TDC).
+
+    Returns
+    -------
+    xf : np.array
+        X front position (hundredths of a millimeter).
+    """
+    # Left and right start types.
+    indices = np.where((start_type == 1) | (start_type == 2))
```
Comment (tmplummer) [inline imap_processing/ultra/l1b/ultra_l1b_extended.py]: numpy documentation suggests the use of `np.nonzero()` when used to get indices like here. The below use of `where` with all three arguments supplied is my take on what the documentation says is the intended use.

File: imap_processing/ultra/l1b/ultra_l1b_extended.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    Returns
+    -------
+    d : np.array
+        Distance front to back in hundredths of a millimeter.
+    yf : np.array
+        Front y position in hundredths of a millimeter.
+    """
+    # Determine start types
+    index_left = np.where(start_type == 1)
+    index_right = np.where(start_type == 2)
```
Comment (tmplummer) [inline imap_processing/ultra/l1b/ultra_l1b_extended.py]: numpy documentation suggests using `np.nonzero()` for getting indices. 
```suggestion
    index_left = np.nonzero(start_type == 1)
    index_right = np.nonzero(start_type == 2)
```

File: imap_processing/ultra/l1b/l1b_extended_config.py
Comment (tmplummer) [inline imap_processing/ultra/l1b/l1b_extended_config.py:1]: Are these config values useful outside of the `ultra_l1b_extended` module? If not, I would suggest just defining them within that module. If so, should they be in a more generic location?

**Review by tmplummer — APPROVED** [review]
Overall, looks good. I pointed out a few nit picks in comments. Thanks for breaking these changes out!

**Pull request merged**